client/alloc_runner.go

```diff
@@ -494,6 +489,19 @@ func (r *AllocRunner) StatsReporter() AllocStatsReporter {
 	return r
 }
 
+// getTaskRunners is a helper that returns a copy of the task runners list using
+// the taskLock.
+func (r *AllocRunner) getTaskRunners() []*TaskRunner {
+	// Get the task runners
+	r.taskLock.RLock()
+	defer r.taskLock.RUnlock()
+	runners := make([]*TaskRunner, 0, len(r.tasks))
+	for _, tr := range r.tasks {
+		runners = append(runners, tr)
```

Comment:
Since we know how many task runners we are going to return and not filtering them
```
runners := make([]*TaskRunner,len(r.tasks))
for i, tr := range r.tasks {
  runners[i] = tr
}
```